docs/writing-tests/test-coverage.mdx

```diff
@@ -5,19 +5,222 @@ sidebar:
+        with:
+          node-version: '20.x'
+      - name: Install dependencies
+        run: yarn
+      - name: Run Storybook tests
+        run: yarn test-storybook --coverage
+```
+
+For more on testing in CI, see the [Test addon docs](./test-addon.mdx#ci).
+
+### Configuration
+
+#### Coverage provider
+
+You can choose which provider, [v8](https://v8.dev/blog/javascript-code-coverage) (default) or [Istanbul](https://istanbul.js.org/), to use for coverage calculation by setting the `coverage.provider` option in your Vitest config:
+
+```ts title="vitest.config.ts"
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  // ...
+  test: {
+    // ...
+    coverage: {
+      // ...
+      provider: 'istanbul', // 'v8' is the default
+    },
+  },
+});
+```
+
+#### Watermarks
+
+Both coverage providers support [watermarks](https://vitest.dev/config/#coverage-watermarks), which are threshold values for coverage. The low watermark is the minimum coverage required to pass the test, and the high watermark is the minimum coverage required to be considered good. A coverage percentage between the low and high watermarks will be considered acceptable but not ideal.
 
-Storybook provides an official [test coverage addon](https://storybook.js.org/addons/@storybook/addon-coverage). Powered by [Istanbul](https://istanbul.js.org/), which allows out-of-the-box code instrumentation for the most commonly used frameworks and builders in the JavaScript ecosystem.
+In the testing module, the coverage summary will show the percentage of statements covered by your tested stories, as well as whether the coverage meets the watermarks. Below the low watermark, the icon will be red, between the low and high watermarks, it will be orange, and above the high watermark, it will be green.
 
-### Set up the coverage addon
+![Screenshot of test module, expanded, showing coverage result](../_assets/writing-tests/addon-test-module-coverage-summary.png)
+
+To configure the watermarks, you can adjust the Vitest config:
+
+```ts title="vitest.config.ts"
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  // ...
+  test: {
+    // ...
+    coverage: {
+      // ...
+      watermarks: {
+        // These are the default values
+        statements: [50, 80],
+      },
+    },
+  },
+});
+```
+
+#### Additional configuration
+
+You can find more configuration options for coverage in the [Vitest documentation](https://vitest.dev/config/#coverage).
+
+When calculating coverage in the Storybook UI, the following options are always ignored:
+
+- `enabled`
+- `clean`
+- `cleanOnRerun`
+- `reportOnFailure`
+- `reporter`
+- `reportsDirectory`
+
+### Troubleshooting
+
+#### Excluding stories from the coverage report
+
+Until Vitest 2.2.0 is released, the generated coverage report will include the stories files themselves. This is misleading and they should be excluded. To do this, you can add the following to your Vitest config:
```

Comment:
`3.0.0`, not `2.2.0`, as they scrapped that minor in favor of a major release instead. See https://github.com/vitest-dev/vitest/releases/tag/v3.0.0-beta.1